icecream/icecream.py

```diff
@@ -154,15 +155,38 @@ def format_pair(prefix, arg, value):
     return '\n'.join(lines)
 
 
-def argumentToString(obj):
-    s = DEFAULT_ARG_TO_STRING_FUNCTION(obj)
+def argumentToString(obj, width=DEFAULT_LINE_WRAP_WIDTH):
+    s = DEFAULT_ARG_TO_STRING_FUNCTION(obj, width=width)
     s = s.replace('\\n', '\n')  # Preserve string newlines in output.
     return s
 
 
+def detect_terminal_width(prefix, default=DEFAULT_LINE_WRAP_WIDTH):
+    """ Returns the number of columns that this terminal can handle. """
+    width = default
+    try:
+        if hasattr(shutil, "get_terminal_size"):
```

Comment:
I'd prefer to have near the top of the file:

```
try:
    from shutil import get_terminal_size
except ImportError:
    try:
        from backports.shutil_get_terminal_size import get_terminal_size
    except ImportError:
        def get_terminal_size():
            # get COLUMNS environment variable
```

icecream should still work if `backports.shutil_get_terminal_size` isn't installed, e.g. if the user chooses to uninstall it even after installing it automatically.

Reply:
Will do it that way then 👍

Reply:
(Although in my terminal, `env['COLUMNS']` isn't set, so I'm not sure which environments/OSes do set that variable)

Reply:
I don't know anything about this either, but I imagine it's a convention so that humans can easily set COLUMNS=50 when running a program and expect that many parts of the program will automatically respect that.